apps/ex_cucumber/lib/ex_cucumber.ex

```diff
@@ -188,9 +198,22 @@ defmodule ExCucumber do
 
               s = List.flatten([left, middle, right])
               reraise e, s
+          catch
+            _ ->
+              callback_on_error(ctx)
+
+            _, _ ->
+              callback_on_error(ctx)
           end
         end
       end
+
+      defp callback_on_error(context) do
+        if Module.has_attribute?(__MODULE__, :on_error) do
+          on_error = Module.get_attribute(__MODULE__, :on_error)
+          if is_function(on_error, 1), do: on_error.(context)
+        end
+      end
```

Comment:
Do we also want to put a `try` ... `rescue` / `catch`  block around the user-supplied callback functions to ensure we also catch errors raised there and handle them appropriately?

Reply:
We might also want to add a warning if the module attribute is not a one-arity function.

```elixir
defp callback_on_error(context) do
  case Module.get_attribute(__MODULE__, :on_error) do
    nil  -> 
      :ok

    on_error when is_function(on_error, 1) -> 
      on_error.(context)

    invalid -> 
      raise ArgumentError, message: "invalid `on_error` module attribute in: ` <> inspect(__MODULE__)
  end
end
```